src/Cooked/MockChain/UtxoState.hs

```diff
@@ -2,47 +2,45 @@ module Cooked.MockChain.UtxoState
   ( UtxoState (..),
     UtxoPayloadSet (..),
     UtxoPayload (..),
-    holdingInState,
+    holdsInState,
   )
 where
 
 import Cooked.Skeleton (TxSkelOutDatum)
-import Cooked.Wallet (Wallet, walletAddress)
 import Data.Function (on)
 import qualified Data.List as List
 import Data.Map.Strict (Map)
 import qualified Data.Map.Strict as Map
 import qualified Ledger as Pl hiding (Value)
-import qualified Plutus.V1.Ledger.Value as Pl1
+import qualified Plutus.Script.Utils.Value as Pl
+import qualified Plutus.V2.Ledger.Api as PV2
 
--- TODO Migrate to V2 values (see imports)
-
--- | A 'UtxoState' provides us with the mental picture of the state of the UTxO graph:
--- Each address has a set of UTxOs that consist in a value and some potential datum.
+-- | A description of who owns what in a blockchain. Owners are addresses
+-- and they each own a 'UtxoPayloadSet'.
 newtype UtxoState = UtxoState {utxoState :: Map Pl.Address UtxoPayloadSet}
   deriving (Eq)
 
--- | Helper function to compute what the given wallet owns in the
--- given state
-holdingInState :: UtxoState -> Wallet -> Pl1.Value
-holdingInState (UtxoState m) w
-  | Just vs <- Map.lookup (walletAddress w) m = utxoPayloadSetTotal vs
-  | otherwise = mempty
+-- | Returns the total value accessible to the entity designed by the
+-- address.
+holdsInState :: PV2.Address -> UtxoState -> PV2.Value
+holdsInState address (UtxoState m) =
+  maybe mempty utxoPayloadSetTotal (Map.lookup address m)
 
 instance Semigroup UtxoState where
   (UtxoState a) <> (UtxoState b) = UtxoState $ Map.unionWith (<>) a b
 
--- | Represents a /set/ of payloads. Payloads are the name we give to the
--- information we care about on a UTxO: value, datum, and reference script. We
--- use a list instead of a set because 'Pl.Value' doesn't implement 'Ord' and
--- because it is possible that we want to distinguish between utxo states that
--- have additional utxos, even if these could have been merged together.
+-- | Represents a /set/ of payloads.
 newtype UtxoPayloadSet = UtxoPayloadSet {utxoPayloadSet :: [UtxoPayload]}
   deriving (Show)
 
+-- We use a list instead of a set because 'Pl.Value' doesn't implement 'Ord'
+-- and because it is possible that we want to distinguish between utxo states
+-- that have additional utxos, even if these could have been merged together.
```

Comment:
Why is it no longer part of the `UtxoPayloadSet` documentation comment?

Reply:
I agree but it seems to be floating in the middle of nowhere in the module, it is not clear what it refers to at first glance. Maybe we could put it near the accessor?

Reply:
Oh it was ormolu mischief. Yes I think it is good now!